homeassistant/components/geniushub/water_heater.py

```diff
@@ -0,0 +1,141 @@
+    'other': None,
+}
+
+GH_DEVICE_STATE_ATTRS = ['type', 'override']
+
+
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Genius Hub water_heater entities."""
+    client = hass.data[DOMAIN]['client']
+
+    entities = [GeniusWaterHeater(client, z)
+                for z in client.hub.zone_objs if z.type in GH_WATER_HEATERS]
+
+    async_add_entities(entities)
+
+
+class GeniusWaterHeater(WaterHeaterDevice):
+    """Representation of a Genius Hub water_heater device."""
+
+    def __init__(self, client, zone):
+        """Initialize the water_heater device."""
+        self._client = client
+        self._objref = zone
+        self._id = zone.id
+        self._name = zone.name
+
+        self._operation_list = list(HA_OPMODE_TO_GH)
+
+    @property
+    def name(self):
+        """Return the name of the water_heater device."""
+        return self._objref.name
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        tmp = self._objref.__dict__.items()
+        state = {k: v for k, v in tmp if k in GH_DEVICE_STATE_ATTRS}
+
+        return {'status': state}
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._objref.temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._objref.setpoint
+
+    @property
+    def min_temp(self):
+        """Return max valid temperature that can be set."""
+        return GENIUSHUB_MIN_TEMP
+
+    @property
+    def max_temp(self):
+        """Return max valid temperature that can be set."""
+        return GENIUSHUB_MAX_TEMP
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return GENIUSHUB_SUPPORT_FLAGS
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return GH_STATE_TO_HA.get(self._objref.mode)
+
+    async def async_set_operation_mode(self, operation_mode):
+        """Set a new operation mode for this zone."""
+        await self._objref.set_mode(HA_OPMODE_TO_GH.get(operation_mode))
+
+    async def async_set_temperature(self, **kwargs):
+        """Set a new target temperature for this zone."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        await self._objref.set_override(temperature, 3600)  # 1 hour
+
+    async def async_update(self):
+        """Get the latest data from the hub."""
+        try:
+            await self._objref.update()
+        except (AssertionError, asyncio.TimeoutError) as err:
```

Comment:
For curiosity, what is asserted?

Reply:
Is basically:
```python
assert response.status == HTTP_OK  # 200
```
I was thinking of changing it to `raise_for_status()`, but will do that in a later PR.

Reply:
`raise_for_status` should be done in the client library. The library should preferably raise its own exceptions, with one common base exception unique for the library.